### ISSUE TITLE
1.0: Constant Transition rejection error?

### ISSUE THREAD

**Issue by ako977** [angular-ui/ui-router#3196]
recently migrated from 0.2.x to 1.0.beta.3 in an angular 1.x app, and for some reason I cannot transition from any state other than the homepage state of the single page application. I am repeatedly getting a TransitionRejection error thrown by ui router.

I know it could be anything, what I would like to find out is **_how to debug what is aborting the state transition_**.
is there any way to determine this? I tried hooking into $transitions service's onStart and onSuccess, but it doesn't show any other state transition that occurred before the one I attempt to invoke.

The error thrown is:
`TransitionRejection(type: 3, message: The transition has been aborted, detail: Hook aborted transition)`

**Comment by christopherthielen** [comment]
1. Enable trace. .run($trace => $trace.enable(1))

2. Check your `onEnter` and `onExit`. They are now hooks and their return value is processed. 

More info here: https://ui-router.github.io/guide/ng1/migrate-to-1_0

**Comment by christopherthielen** [comment]
3) Put a breakpoint in the [defaultErrorHandler](https://github.com/ui-router/core/blob/b64973694e77c85ff331b51fc9d10ff8e975b4fb/src/state/stateService.ts#L535)

**Comment by ako977** [comment]
Thanks @christopherthielen , option 1. was the best.
Reading the docs on "trace" I configured it with 'viewconfig' to even see which views were being instantiated:
`$trace.enable("TRANSITION", "VIEWCONFIG");`

**Comment by ako977** [comment]
Just an FYI for others who may encounter this, the trace helped, but also did not record the transition being aborted. This however, did help me verify that something else was indeed preventing the state from even being started. The next step was to to monitor other code that may intercept/crash routes at a lower level from UI Router or in modules before your app is run.

In this case:
`$rootScope.$on('$locationChangeStart', function () { ... `
or
`$rootScope.$on('$routeChangeStart', function () { ... `

**Comment by haoyongliang** [comment]
I'm using angular version 1.6, angular-ui-router is 1.0.4, but the problem still exists

**Comment by Baxterboom** [comment]
same here for 1.0.5 and using (stateEvents.js) and Angular 1.5.11

**Comment by GauthamBanasandra** [comment]
I faced this issue. In my controller, I had written a handler which would open a modal dialog for selecting a file and upon doing so, it would transition to some state. I was using -
```javascript
$state.go('app.state.1', {
    param: 'some parameter'
});
```
I used to get this error sometimes -
```json
{$id: 0, type: 3, message: "The transition has been aborted", detail: "Hook aborted transition"}
```

I went through the docs and came across `[$state.transitionTo](https://github.com/angular-ui/ui-router/wiki/Quick-Reference#statetransitiontoto-toparams--options)` and found that `$state.go` calls `$state.transitionTo` with some default parameters.
One of the default parameters that `$state.go` passes is `reload:false`. I changed to
```javascript
$state.transitionTo('app.state.1', {
    param: 'some parameter'
}, {
    reload : true
});
```
And that error went away.
